test/units/TransferAdminRoleScript.t.sol

```diff
@@ -0,0 +1,204 @@
+        acceptAdminScript = new TestableAcceptAdminRole();
+
+        // Label contracts for better debugging
+        vm.label(address(rlcLiquidityUnifier), "RLCLiquidityUnifier");
+        vm.label(address(rlcCrosschainToken), "RLCCrosschainToken");
+        vm.label(admin, "admin");
+        vm.label(newAdmin, "newAdmin");
+    }
+
+    // ====== revert scenarios checks ======
+    function test_RevertWhen_NewAdminIsZeroAddress() public {
+        vm.startPrank(admin);
+        vm.expectRevert("BeginTransferAdminRole: new admin cannot be zero address");
+        beginTransferScript.publicBeginTransfer(address(rlcLiquidityUnifier), address(0), "RLCLiquidityUnifier");
+        vm.stopPrank();
+    }
+
+    function test_RevertWhen_NewAdminIsSameAsCurrentAdmin() public {
+        vm.startPrank(admin);
+        vm.expectRevert("BeginTransferAdminRole: New admin must be different from current admin");
+        beginTransferScript.publicBeginTransfer(address(rlcLiquidityUnifier), admin, "RLCLiquidityUnifier");
+        vm.stopPrank();
+    }
+
+    function test_RevertWhen_NotAuthorizedToTransferAdmin() public {
+        address unauthorizedUser = makeAddr("unauthorizedUser");
+        vm.startPrank(unauthorizedUser);
+        vm.expectRevert(); // Should revert with access control error
+        beginTransferScript.publicBeginTransfer(address(rlcLiquidityUnifier), newAdmin, "RLCLiquidityUnifier");
+        vm.stopPrank();
+    }
+
+    function test_RevertWhen_WrongAddressTriesToAcceptAdmin() public {
+        beginTransferScript.publicBeginTransferAsAdmin(
+            address(rlcLiquidityUnifier), newAdmin, "RLCLiquidityUnifier", admin
+        );
+
+        // Try to accept with wrong address using the script wrapper
+        address wrongAddress = makeAddr("wrongAddress");
+
+        vm.expectRevert(); // Should revert because only pending admin can accept
+        acceptAdminScript.publicAcceptContractAdminAsUser(
+            address(rlcLiquidityUnifier), "RLCLiquidityUnifier", wrongAddress
+        );
+    }
+
+    // ====== BeginTransferAdminRole.validateAdminTransfer ======
+    function test_ValidateAdminTransfer() public {
+        // Test the validation function directly
+        vm.startPrank(admin);
+
+        // Should not revert with valid inputs
+        beginTransferScript.publicValidateAdminTransfer(admin, newAdmin);
+
+        // Should revert with zero address
+        vm.expectRevert("BeginTransferAdminRole: new admin cannot be zero address");
+        beginTransferScript.publicValidateAdminTransfer(admin, address(0));
+
+        // Should revert when new admin is same as current
+        vm.expectRevert("BeginTransferAdminRole: New admin must be different from current admin");
+        beginTransferScript.publicValidateAdminTransfer(admin, admin);
+
+        vm.stopPrank();
+    }
+
+    // ====== BeginTransferAdminRole.beginTransfer ======
+    function test_BeginTransferAdminRole_LiquidityUnifier() public {
```

Comment:
We don't need to test `beginTransfer` function with all contracts. 1 test is enough.
Also we need to check that the script sends request for all contracts.